include/swift/Basic/BasicBridging.h

```diff
@@ -25,6 +25,7 @@
 
 #include <stddef.h>
 #include <stdint.h>
+#include <vector>
```

Comment:
Please don't add any C++ headers in BasicBridging.h

Reply:
Is it even needed to bridge a std::vector of BridgedCharSourceRange?

If this vector is constructed on the swift side and then passed to C++, you can just use a swift Array and pass its buffer to C++, like done here: https://github.com/apple/swift/blob/b90b3261164f149030243b856f5dcfce12a34662/SwiftCompilerSources/Sources/SIL/Utils.swift#L224

If it's constructed on the C++ side and used on the swift side, you can bridge an ArrayRef like struct.